### PR TITLE
[KAFKA-4559] Add a site search bar on the Web site

### PR DESCRIPTION
**Pull request by zer0Id0l**
What?
Providing search functionality for **Kafka** Documentation by integrating with [Algolia Docsearch](https://community.algolia.com/docsearch/).

Why?
As developers, we spend a lot of time reading documentation, and it isn’t always easy to find the information we need. This pull request will help developers find the needed content easily.

Since Docsearch handles the crawling of webpage and also configuring the search experience, its very maintainable too.
 
This pull request has code to adds the `html`, `js` and `css` required to enable the search.

Special thanks to **@s-pace** from **Algolia** who helped in configuring the search.

### CODE REVIEW

File: includes/_header.htm [hunk 1]
```diff
@@ -56,4 +57,17 @@
...
+
+
+			<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js"></script>
+			<script type="text/javascript">
+				document.addEventListener("DOMContentLoaded", function(event) {
+					docsearch({
+					apiKey: 'aa09166cb320ff3715048b35f14631ca',
+					indexName: 'kafka_apache',
+					inputSelector: '#searchbox',
+					debug: true // Set debug to true if you want to inspect the dropdown
```
Comment (s-pace) [inline includes/_header.htm]: @zer0Id0l You can set this boolean to false in order to automatically hide the dropdown. It will stay open otherwise.

**Comment by s-pace** [comment]
Happy to help here.

💯 @zer0Id0l

**Comment by s-pace** [comment]
We do provide analytics regarding the usage of your search.

We can also provide you a way to filter on versions in order to handle the whole of them.

Feel free to ping me.

File: includes/_header.htm (hunk 1 above)
Comment (zer0Id0l) [reply includes/_header.htm]: Made the changes. Thanks!

**Comment by zer0Id0l** [comment]
@guozhangwang Please let us know if it would be better to have search based on different Kafka versions. 
Current implementation searches with in the latest documentation.

**Comment by guozhangwang** [comment]
@manjuapu @derrickdoo could you help reviewing this PR? The goal was to add a site-scope search bar on AK docs.

**Comment by zer0Id0l** [comment]
@manjuapu @derrickdoo Can you please help reviewing this PR?

**Comment by zer0Id0l** [comment]
@guozhangwang Can you please review? Looks like manjuapu and derrickdoo are busy.

**Comment by zer0Id0l** [comment]
@guozhangwang @derrickdoo @manjuapu

**Comment by zer0Id0l** [comment]
@guozhangwang @derrickdoo @manjuapu Checking if someone can still review this.

**Review by s-pace — APPROVED** [review]

**Comment by s-pace** [comment]
@zer0Id0l I have just approved this PR to let you know that everything was fine from our end. 

DISCLAIMER: I am not a maintainer of this repo.

**Comment by mjsax** [comment]
Closing this PR as abandoned. (We can reopen if necessary.)

**Pull request closed without merging**